076-connector-offsets-support.md

```diff
@@ -0,0 +1,286 @@
+  ```yaml
+  - lastTransitionTime: "2024-06-04T08:44:15.913138115Z"
+    message: Failed to alter the connector offsets due to "message from endpoint".
+    reason: ResetOffsets
+    status: "True"
+    type: Warning
+  ```
+The operator will leave the annotation on the KafkaConnector resource until either the delete operation succeeds, or the user removes the annotation.
+This means the operator will retry the delete on every operation, allowing the condition to remain present for the user to see.
+
+Strimzi will shortcut and automatically fail to do the reset if the KafkaConnector resource does not have it's `state` set as `stopped`.
+Similar to if Connect returns on error, the operator will add a condition stating that the operation has failed because the connector is not stopped and therefore offsets cannot be reset, and leave the annotation on the resource.
+The user can update the KafkaConnector to stop the connector and reset the offsets at the same time.
+In that case the operator will first stop the connector, and once that API call returns, then it will make the call to reset the offsets.
+
+## Affected/not affected projects
+
+This proposal only affects the Connect parts of the cluster-operator.
+This proposal does not apply to the use of the KafkaMirrorMaker2 Custom Resource.
+If users are using a KafkaConnector Custom Resource to manage the Mirror Maker connectors, they can manage offsets using the method described in this proposal.
+We should have a different proposal to cover how we might support managing offsets for Mirror Maker connectors that are managed via the KafkaMirrorMaker2 Custom Resource.
+This is because the structure of the offsets JSON object for these connectors is known, so Strimzi can provide a more guided experience for altering offsets.
+For example, by allowing the user to specify the value of specific fields in the offset JSON, rather than them having to construct the full JSON object themselves.
+
+## Compatibility
+
+N/A
+
+## Rejected alternatives
+
+### Listing offsets continually
+A previous version of this proposal suggested to have a property that prompted the operator to add the current offsets to the KafkaConnector status on every reconciliation.
+This was rejected because it will be tricky to implement this in a way that doesn't result in cyclical reconciliations.
+In addition, it seems likely that listing offsets will be used most often as a one-off action, rather than users needing to continually track offsets.
+
+### Listing offsets in the KafkaConnector status
+
+A previous version of this proposal suggested using the KafkaConnector status field to list the returned offsets.
+This was rejected because updating the status would trigger an addition reconcile and the response from the endpoint may be a very large size if there are many partitions.
```

Comment:
```suggestion
This was rejected because updating the status would trigger an additional reconcile and the response from the endpoint may be a very large size if there are many partitions.
```